src/code/z_parameter.c

```diff
@@ -1295,25 +1387,561 @@ TexturePtr D_801BFCC4[] = {
+                            play->nextEntrance = ENTRANCE(TERMINA_FIELD, 1);
+                            gSaveContext.nextCutsceneIndex = 0xFFF0;
+                            play->transitionTrigger = TRANS_TRIGGER_START;
+                        } else {
+                            Interface_StartMoonCrash(play);
+                        }
+                    } else if (gSaveContext.timerStates[TIMER_ID_GORON_RACE_UNUSED] != TIMER_STATE_OFF) {
+                        gSaveContext.timerX[TIMER_ID_GORON_RACE_UNUSED] = 115;
+                        gSaveContext.timerY[TIMER_ID_GORON_RACE_UNUSED] = 80;
+                        if (gSaveContext.timerStates[TIMER_ID_GORON_RACE_UNUSED] <= TIMER_STATE_10) {
+                            gSaveContext.timerStates[TIMER_ID_GORON_RACE_UNUSED] = TIMER_STATE_MOVING_TIMER;
+                        }
+                    }
+                    break;
+
+                case TIMER_STATE_6:
+                    osTime = osGetTime();
+
+                    gSaveContext.timerStopTimes[sTimerId] =
+                        OSTIME_TO_TIMER(osTime - ((void)0, gSaveContext.timerStartOsTimes[sTimerId]) -
+                                        ((void)0, gSaveContext.timerPausedOsTimes[sTimerId]));
+
+                    if ((gSaveContext.minigameState == 1) &&
+                        (gSaveContext.save.entrance == ENTRANCE(ROMANI_RANCH, 0))) {
+                        if (gSaveContext.timerStopTimes[sTimerId] >= SECONDS_TO_TIMER(120)) {
+                            gSaveContext.timerStopTimes[sTimerId] = SECONDS_TO_TIMER(120);
+                            gSaveContext.timerCurTimes[sTimerId] = SECONDS_TO_TIMER(120);
+                        }
+                    } else if ((gSaveContext.eventInf[3] & 0x10) && (play->sceneNum == SCENE_DEKUTES) &&
+                               (gSaveContext.timerStopTimes[sTimerId] >= SECONDS_TO_TIMER(120))) {
+                        gSaveContext.timerCurTimes[sTimerId] = SECONDS_TO_TIMER(120);
+                    }
+                    gSaveContext.timerStates[sTimerId] = TIMER_STATE_7;
+
+                    if (gSaveContext.timerStates[TIMER_ID_GORON_RACE_UNUSED] != TIMER_STATE_OFF) {
+                        gSaveContext.timerX[TIMER_ID_GORON_RACE_UNUSED] = 115;
+                        gSaveContext.timerY[TIMER_ID_GORON_RACE_UNUSED] = 80;
+                        if (gSaveContext.timerStates[TIMER_ID_GORON_RACE_UNUSED] <= TIMER_STATE_10) {
+                            gSaveContext.timerStates[TIMER_ID_GORON_RACE_UNUSED] = TIMER_STATE_MOVING_TIMER;
+                        }
+                        gSaveContext.timerStates[sTimerId] = TIMER_STATE_OFF;
+                    }
+                    break;
+            }
+            break;
+        }
+
+        // Update timer counting
+        if ((sTimerId != TIMER_ID_NONE) && gSaveContext.timerStates[sTimerId]) { // != TIMER_STATE_OFF
```

Comment:
I think it would leave a lot of bloat everywhere to add a matching comment, especially since so many of these are same-line comments, but it's open to discussion

Reply:
Yeah, we need some other opinions here since it's a significant amount of code it affects.